### PR TITLE
Fix bug: Terminal logger project output path link (localization)

### LINKED ISSUE #9675: [NETSDKE2E]The path which contains loc characters cannot be opened via "Ctrl + Click" in new terminal logger output in CLI on Loc OS.
### Issue Description

The path which contains loc characters cannot be entered in new terminal logger output in CLI on Loc OS. 

### Steps to Reproduce

Affect build:
.NET 9.0.100-alpha.1.24072.8 from [dotnet/installer: .NET SDK Installer (github.com)](https://github.com/dotnet/installer)
Affected Product Language:
All Loc OS

1.Create a project named with loc characters in CLI.
dotnet new console -o 本地化
2. cd 本地化
3.dotnet build
4. Access to the path "bin\Debug\net9.0\本地化.dll"

### Expected Behavior

The path can be opened successfully.

### Actual Behavior

cannot open the path.

Chinese OS:
image

Turkish OS:
image

EN(not repro):
image

dotnet --info:
.NET SDK:
Version: 9.0.100-alpha.1.24072.8
Commit: ac061bfeca
Workload version: 9.0.100-manifests.fd1572dc

运行时环境:
OS Name: Windows
OS Version: 10.0.22621
OS Platform: Windows
RID: win-x64
Base Path: C:\Program Files\dotnet\sdk\9.0.100-alpha.1.24072.8\

已安装 .NET 工作负载:
没有要显示的已安装工作负载。

Host:
Version: 9.0.0-alpha.1.24071.3
Architecture: x64
Commit: 8accd8073a

.NET SDKs installed:
9.0.100-alpha.1.24072.8 [C:\Program Files\dotnet\sdk]

.NET runtimes installed:
Microsoft.AspNetCore.App 9.0.0-alpha.1.24069.6 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
Microsoft.NETCore.App 9.0.0-alpha.1.24071.3 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
Microsoft.WindowsDesktop.App 9.0.0-alpha.1.24071.1 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]

Other architectures found:
None

Environment variables:
Not set

global.json file:
Not found

Learn more:
https://aka.ms/dotnet/info

Download .NET:
https://aka.ms/dotnet/download

Analysis

### Analysis

_No response_

### Versions & Configurations

_No response_

### PR DESCRIPTION
**Pull request by surayya-MS**
Fixes #9675

### Bug:

TerminalLogger_ProjectOutputPath_Bug

### Fix:

TerminalLogger_ProjectOutputPath_Fix

### CODE REVIEW

File: src/MSBuild/TerminalLogger/TerminalLogger.cs [hunk 1]
```diff
@@ -645,7 +645,7 @@ private void ProjectFinished(object sender, ProjectFinishedEventArgs e)
                             string urlString = url.ToString();
                             if (Uri.TryCreate(urlString, UriKind.Absolute, out Uri? uri))
                             {
-                                urlString = uri.AbsoluteUri;
+                                urlString = uri.ToString();
```
Comment (rainersigwald) [inline src/MSBuild/TerminalLogger/TerminalLogger.cs:648]: I think this deserves a comment: why does the other API mangle it? Should we not use this `Uri.TryCreate` at all?

File: src/MSBuild.UnitTests/TerminalLogger_Tests.cs [hunk 2]
```diff
@@ -38,6 +38,7 @@ public class TerminalLogger_Tests : IEventSource, IDisposable
 
         private readonly string _projectFile = NativeMethods.IsUnixLike ? "/src/project.proj" : @"C:\src\project.proj";
         private readonly string _projectFile2 = NativeMethods.IsUnixLike ? "/src/project2.proj" : @"C:\src\project2.proj";
+        private readonly string _projectFileWithForeignSymbols = NativeMethods.IsUnixLike ? "/src/проектТерминал/本地化.proj" : @"C:\src\проектТерминал\本地化.proj";
```
Comment (rainersigwald) [inline src/MSBuild.UnitTests/TerminalLogger_Tests.cs:41]:
```suggestion
        private readonly string _projectFileWithNonAnsiSymbols = NativeMethods.IsUnixLike ? "/src/проектТерминал/本地化.proj" : @"C:\src\проектТерминал\本地化.proj";
```

Comment (rainersigwald) [inline src/MSBuild.UnitTests/TerminalLogger_Tests.cs:41]: Should we add any of the GB18030 PUA characters here?

File: src/MSBuild.UnitTests/TerminalLogger_Tests.cs [hunk 3]
```diff
@@ -233,20 +234,20 @@ private BuildErrorEventArgs MakeErrorEventArgs(string error)
 
         #region Build summary tests
 
-        private void InvokeLoggerCallbacksForSimpleProject(bool succeeded, Action additionalCallbacks)
+        private void InvokeLoggerCallbacksForSimpleProject(bool succeeded, string projectFile, Action additionalCallbacks)
         {
```
Comment (rainersigwald) [inline src/MSBuild.UnitTests/TerminalLogger_Tests.cs:238]: Consider a maybe-more-ergonomic change

```suggestion
        private void InvokeLoggerCallbacksForSimpleProject(bool succeeded, Action additionalCallbacks, string projectFile = null)
        {
            projectFile ??= _projectFile;
```

(then you wouldn't have to make the other callsites more complicated)

File: src/MSBuild.UnitTests/TerminalLogger_Tests.cs [hunk 4]
```diff
@@ -449,6 +450,22 @@ public Task PrintBuildSummary_2Projects_FailedWithErrorsAndWarnings()
             return Verify(_outputWriter.ToString(), _settings).UniqueForOSPlatform();
         }
 
+        [Fact]
+        public Task PrintProjectOutputDirectoryLink()
+        {
+            // Send message in order to set project output path
+            BuildMessageEventArgs e = MakeMessageEventArgs(
+                    $"本地化 -> {_projectFileWithForeignSymbols.Replace("cproj", "dll")}",
```
Comment (rainersigwald) [inline src/MSBuild.UnitTests/TerminalLogger_Tests.cs:458]: It doesn't contain `cproj` now, only `.proj`--intentional?

**Review by rainersigwald — APPROVED** [review]

File: src/MSBuild.UnitTests/TerminalLogger_Tests.cs (hunk 4 above)
Comment (surayya-MS) [reply src/MSBuild.UnitTests/TerminalLogger_Tests.cs:458]: Yes, in order to reflect real scenario. But it really can be anything, won't affect the test

Comment (rainersigwald) [reply src/MSBuild.UnitTests/TerminalLogger_Tests.cs:458]: What I'm saying is that this `Replace()` call doesn't do anything. Did you want it to?

File: src/MSBuild/TerminalLogger/TerminalLogger.cs (hunk 1 above)
Comment (surayya-MS) [reply src/MSBuild/TerminalLogger/TerminalLogger.cs:648]: I agree, I'll add the comment.
Here is what i found 
 https://github.com/Azure/azure-sdk-for-net/issues/12102

File: src/MSBuild.UnitTests/TerminalLogger_Tests.cs (hunk 4 above)
Comment (surayya-MS) [reply src/MSBuild.UnitTests/TerminalLogger_Tests.cs:458]: Oh, sorry read the comment wrong.
It is a typo, should be proj as well. Thanks!

File: src/MSBuild.UnitTests/TerminalLogger_Tests.cs (hunk 2 above)
Comment (surayya-MS) [reply src/MSBuild.UnitTests/TerminalLogger_Tests.cs]: done

File: src/MSBuild/TerminalLogger/TerminalLogger.cs (hunk 1 above)
Comment (surayya-MS) [reply src/MSBuild/TerminalLogger/TerminalLogger.cs:650]: done

**Review by AR-May — APPROVED** [review]
LGTM

**Pull request merged**